### ISSUE TITLE
lid_driven_cavity & thermal_bubble: ERROR: LoadError: ArgumentError: length(size) must be 2.

### ISSUE THREAD

**Issue by maeckha** [CliMA/Oceananigans.jl#1507]
Hello,
maybe this is an easy one since those mentioned scripts ([lid_driven_cavity](https://github.com/CliMA/Oceananigans.jl/blob/master/validation/lid_driven_cavity/lid_driven_cavity.jl) & [thermal_bubble](https://github.com/CliMA/Oceananigans.jl/blob/master/validation/thermal_bubble/thermal_bubble.jl)) don't seem that complicated. But after several days of reading code, poking around and trying stuff out I couldn't get rid of this Error when running them.
Did someone have or can point me to a solution?

**Comment by navidcy** [comment]
Hi @maeckha, could you post exactly what you try to run and what the error is? Also the output of 
```julia
julia> using Pkg; Pkg.status()
```

**Comment by maeckha** [comment]
Sure!

```
(Oceananigans) pkg> status
Project Oceananigans v0.53.0
Status `~/Documents/Bachelor Arbeit/Repos/Oceananigans.jl/Project.toml`
  [79e6a3ab] Adapt v3.2.0
  [052768ef] CUDA v2.4.1
  [a8cc5b0e] Crayons v4.0.4
  [7a1cc6ca] FFTW v1.3.2
  [e9467ef8] GLMakie v0.1.30
  [c27321d9] Glob v1.3.0
  [033835bb] JLD2 v0.4.3
  [63c18a36] KernelAbstractions v0.5.4
  [da04e1cc] MPI v0.16.1
  [442fdcdd] Measures v0.3.1
  [85f8d34a] NCDatasets v0.11.3
  [6fe1bfb0] OffsetArrays v1.6.2
  [bac558e1] OrderedCollections v1.4.0
  [4a48f351] PencilFFTs v0.12.2
  [d330b81b] PyPlot v2.9.0
  [1bc83da4] SafeTestsets v0.0.1
  [d496a93d] SeawaterPolynomials v0.2.0
  [09ab397b] StructArrays v0.5.0
  [ade2ca70] Dates
  [b77e0a4c] InteractiveUtils
  [37e2e46d] LinearAlgebra
  [56ddb016] Logging
  [44cfe95a] Pkg
  [de0858da] Printf
  [9a3f8284] Random
  [10745b16] Statistics
```

```
ERROR: LoadError: ArgumentError: length(size) must be 2.
Stacktrace:
 [1] validate_tupled_argument(::Tuple{Int64,Int64,Int64}, ::Type{T} where T, ::String, ::Int64; greater_than::Int64) at /home/meck/Documents/Bachelor Arbeit/Repos/Oceananigans.jl/src/Grids/input_validation.jl:24
 [2] validate_tupled_argument(::Tuple{Int64,Int64,Int64}, ::Type{T} where T, ::String, ::Int64) at /home/meck/Documents/Bachelor Arbeit/Repos/Oceananigans.jl/src/Grids/input_validation.jl:24
 [3] validate_size(::Type{T} where T, ::Type{T} where T, ::Type{T} where T, ::Tuple{Int64,Int64,Int64}) at /home/meck/Documents/Bachelor Arbeit/Repos/Oceananigans.jl/src/Grids/input_validation.jl:48
 [4] RegularRectilinearGrid(::Type{T} where T; size::Tuple{Int64,Int64,Int64}, x::Tuple{Int64,Int64}, y::Tuple{Int64,Int64}, z::Tuple{Int64,Int64}, extent::Nothing, topology::Tuple{DataType,DataType,DataType}, halo::Nothing) at /home/meck/Documents/Bachelor Arbeit/Repos/Oceananigans.jl/src/Grids/regular_rectilinear_grid.jl:161
 [5] simulate_lid_driven_cavity(; Re::Int64, N::Int64, end_time::Int64) at /home/meck/Documents/Bachelor Arbeit/Repos/Oceananigans.jl/validation/lid_driven_cavity/lid_driven_cavity.jl:15
 [6] top-level scope at /home/meck/Documents/Bachelor Arbeit/Repos/Oceananigans.jl/validation/lid_driven_cavity/lid_driven_cavity.jl:91
in expression starting at /home/meck/Documents/Bachelor Arbeit/Repos/Oceananigans.jl/validation/lid_driven_cavity/lid_driven_cavity.jl:91
```

So basically there are six errors starting with the lid_driven_cavity at the top level going all the way down to input_validation.

**Comment by navidcy** [comment]
The errors come up when you call Pkg.status()?

it’ll be helpful to see what exactly you try to run and getting these errors.

**Comment by maeckha** [comment]
I copied the output of pgk.status as per your request.

The second part comes out when I am running lid_driven_cavity.jl. I run it in VS Codium.

```
julia> simulate_lid_driven_cavity(Re=100,   N=128, end_time=15)
ERROR: ArgumentError: length(size) must be 2.
Stacktrace:
 [1] validate_tupled_argument(::Tuple{Int64,Int64,Int64}, ::Type{T} where T, ::String, ::Int64; greater_than::Int64) at /home/meck/Documents/Bachelor Arbeit/Repos/Oceananigans.jl/src/Grids/input_validation.jl:24
 [2] validate_tupled_argument(::Tuple{Int64,Int64,Int64}, ::Type{T} where T, ::String, ::Int64) at /home/meck/Documents/Bachelor Arbeit/Repos/Oceananigans.jl/src/Grids/input_validation.jl:24
 [3] validate_size(::Type{T} where T, ::Type{T} where T, ::Type{T} where T, ::Tuple{Int64,Int64,Int64}) at /home/meck/Documents/Bachelor Arbeit/Repos/Oceananigans.jl/src/Grids/input_validation.jl:48
 [4] RegularRectilinearGrid(::Type{T} where T; size::Tuple{Int64,Int64,Int64}, x::Tuple{Int64,Int64}, y::Tuple{Int64,Int64}, z::Tuple{Int64,Int64}, extent::Nothing, topology::Tuple{DataType,DataType,DataType}, halo::Nothing) at /home/meck/Documents/Bachelor Arbeit/Repos/Oceananigans.jl/src/Grids/regular_rectilinear_grid.jl:161
 [5] simulate_lid_driven_cavity(; Re::Int64, N::Int64, end_time::Int64) at /home/meck/Documents/Bachelor Arbeit/Repos/Oceananigans.jl/validation/lid_driven_cavity/lid_driven_cavity.jl:15
 [6] top-level scope at REPL[2]:1

```

**Comment by navidcy** [comment]
@ali-ramadhan is there anything broken in validation exps in v0.53.0?

@maeckha if you try to run some example from the docs is it all ok or similar issues?

**Comment by navidcy** [comment]
@maeckha I can reproduce your error when calling `simulate_lid_driven_cavity()`. I feel that this validation experiment was done with an older version of Oceananigans compared to the v0.53.0 are running and some things are deprecated. @ali-ramadhan might comment more on that perhaps.

But the thermal bubble example from the Readme (just copy paste the code from there) runs just fine! Do you get an error with that also?

**Comment by maeckha** [comment]
Can you show me please the way you are running the thermal bubble example?

**Comment by navidcy** [comment]
I simply paste the code from the README in a Julia REPL.

```Julia
julia> using Oceananigans
[ Info: Oceananigans will use 12 threads

julia> N = Nx = Ny = Nz = 128   # Number of grid points in each dimension.
128

julia> L = Lx = Ly = Lz = 2000  # Length of each dimension.
2000

julia> topology = (Periodic, Periodic, Bounded)
(Periodic, Periodic, Bounded)

julia> model = IncompressibleModel(
           architecture = CPU(),
                   grid = RegularRectilinearGrid(topology=topology, size=(Nx, Ny, Nz), extent=(Lx, Ly, Lz)),
                closure = IsotropicDiffusivity(ν=4e-2, κ=4e-2)
       )
IncompressibleModel{CPU, Float64}(time = 0 seconds, iteration = 0)
├── grid: RegularRectilinearGrid{Float64, Periodic, Periodic, Bounded}(Nx=128, Ny=128, Nz=128)
├── tracers: (:T, :S)
├── closure: IsotropicDiffusivity{Float64,NamedTuple{(:T, :S),Tuple{Float64,Float64}}}
├── buoyancy: SeawaterBuoyancy{Float64,LinearEquationOfState{Float64},Nothing,Nothing}
└── coriolis: Nothing

julia> # Set a temperature perturbation with a Gaussian profile located at the center.
       # This will create a buoyant thermal bubble that will rise with time.
       x₀, z₀ = Lx/2, Lz/2
(1000.0, 1000.0)

julia> T₀(x, y, z) = 20 + 0.01 * exp(-100 * ((x - x₀)^2 + (z - z₀)^2) / (Lx^2 + Lz^2))
T₀ (generic function with 1 method)

julia> set!(model, T=T₀)
julia> simulation = Simulation(model, Δt=10, stop_iteration=5000)
Simulation{IncompressibleModel{CPU, Float64}}
├── Model clock: time = 0 seconds, iteration = 0
├── Next time step (Int64): 10 seconds
├── Iteration interval: 1
├── Stop criteria: Any[Oceananigans.Simulations.iteration_limit_exceeded, Oceananigans.Simulations.stop_time_exceeded, Oceananigans.Simulations.wall_time_limit_exceeded]
├── Run time: 0 seconds, wall time limit: Inf
├── Stop time: Inf years, stop iteration: 5000
├── Diagnostics: OrderedCollections.OrderedDict with 1 entry:
│   └── nan_checker => NaNChecker
└── Output writers: OrderedCollections.OrderedDict with no entries

julia> run!(simulation)
```

How were you trying to run the thermal bubble example?

**Comment by navidcy** [comment]
Hm, I see... you might be trying to run the one found inside the `validation` directory, right?

I suggest you start of by trying to run the scripts in the `examples` directory. They are continuously tested and should run with the latest version.

We'll figure out though the issues and fix the problems for the scripts that live in the `validation` directory (if you are specifically interested on one of those).

**Comment by ali-ramadhan** [comment]
Hi @maeckha sorry for the out-of-date scripts. I've been meaning to add the validation experiments to CI so this doesn't happen.

I can open a PR later today with an updated lid-driven cavity script if that would be helpful.

**Comment by ali-ramadhan** [comment]
@maeckha Ah this got automatically closed by PR #1516 but let us know if the lid-driven cavity script gives you any more troubles (and feel free to re-open).

**Comment by maeckha** [comment]
Thank you all for your work, I will check the new scripts!
